P1371R1.md

```diff
@@ -724,46 +742,133 @@ int get_area(const Shape& shape) {
+ as _binding_ again]{.note}
 
 ```cpp
-std::variant<Point, /* ... */> v = /* ... */;
+enum Color { Red, Green, Blue };
+Color color = /* ... */;
+
+inspect (color) {
+    case Red:   // ...
+    case Green: // ...
+//       ^^^^^ id-expression
+    case Blue:  // ...
+//  ^^^^^^^^^ case pattern
+}
+```
+
+```cpp
+static constexpr int zero = 0;
+int v = /* ... */;
 
 inspect (v) {
-    <Point> ([x, y]): // ...
-//          ^^^^^^^^ parenthesized pattern
+    case zero: std::cout << "got zero";
+//       ^^^^ id-expression
+    case 1: std::cout << "got one";
+//       ^ expression pattern
+    case 2: std::cout << "got two";
+//  ^^^^^^ case pattern
+}
+```
+
+```cpp
+static constexpr int zero = 0, one = 1;
+std::pair<int, int> p = /* ... */
+
+inspect (p) {
+    case [zero, one]: std::cout << zero << ' ' << one;
+//        ^^^^  ^^^ id-expression
 }
 ```
 
 #### Binding Pattern
 
-The binding pattern has the form:
+The binding pattern has the following two forms:
 
-> | _identifier_ = _pattern_
+> | `let` _pattern_
+> | `let` _identifier_ `=` _pattern_
 
-and matches value `v` if _pattern_ matches it. The introduced name behaves as
-an lvalue referring to `v`, and is in scope from its point of declaration until
-the end of the statement following the pattern label.
+Both forms match value `v` if _pattern_ matches `v`.
+For the second form, the _identifier_ creates a binding to `v`.
+
+The top-level pattern is implicitly the first form.
+
+[[@P1371R0] had used `@` for the syntax of the second form.
+ EWG gave clear guidance that this would not be accepted.
+ Other options considered are: `%`, `$`, `:=`, `as` and `by`.]{.note}
+
+```cpp
+int v = /* ... */;
+
+inspect (v) {
+    /* let */ x: std::cout << x;
+//            ^ identifier pattern
+}
+```
+
+```cpp
+static constexpr int two = 2;
+int v = 42;
+
+inspect (v) {
+    let 0: std::cout << "got zero";
+//      ^ expression pattern
+    let 1: std::cout << "got one";
+//  ^^^^^ let pattern
+    let two: std::cout << two;
+//      ^^^ identifier pattern
+}
+
+// prints: 42
+```
+
+```cpp
+static constexpr int zero = 0, one = 1;
+std::pair<int, std::pair<int, int>> p = /* ... */
+
+inspect (p) {
+    case [zero, let [x, y]]: std::cout << "got zero" << ' ' << x << ' ' << y;
+//                   ^  ^ identifier pattern
+    case [one, let [x, y]]: std::cout << "got one" << ' ' << x << ' ' << y;
+//             ^^^^^^^^^^  binding pattern
+}
+```
 
 ```cpp
 std::variant<Point, /* ... */> v = /* ... */;
 
 inspect (v) {
-    <Point> (p = [x, y]): // ...
-//           ^^^^^^^^^^ binding pattern
+    <Point> (let p = [x, y]): // ...
+//           ^^^^^^^^^^^^^^ binding pattern
 }
 ```
 
-Previous versions of this paper used `@` for the binding pattern. EWG gave clear
-feedback that this wouldn't be accepted, and voiced a preference for `=` which
-we have duly adopted. We also considered `%`, `$`, `:=`, `as` and `by` as
-potential alternatives.
+#### Parenthesized Pattern
```

Comment:
Should we move this section up, so that it is before case/let where we use it in examples?